Libraries/LibGUI/ColumnsView.cpp

```diff
@@ -62,6 +62,24 @@ ColumnsView::~ColumnsView()
 {
 }
 
+void ColumnsView::select_all() {
+    Column current_column;
+    for (int i = 0; i < m_columns.size(); i++) {
+        auto& column = m_columns[i];
+        int row_count = model()->row_count(column.parent_index);
+        for (int row = 0; row < row_count; row++) {
+            ModelIndex index = model()->index(row, m_model_column, column.parent_index);
+            if(selection().contains(index)) current_column = column;
+        }
+    }
```

Comment:
Overall, wouldn't it be simpler to look at the selection and find a column with a matching parent index instead of iterating over all items and trying to find which ones are selected?

Reply:
Hmm, I'm not sure how that would look with the current selection API. Do you have an idea how that could be implemented?

Reply:
Like this, for example:

```cpp
selection().for_each_index([&](auto& index) {
    for (auto& column : m_columns) {
        if (column.parent_index == index.parent()) {
            // ... add this column to the new selection...
            return;
        }
    }
    ASSERT_NOT_REACHED();
});

Reply:
Ah, index.parent() is what I was missing. Thank you!